card/src/main/java/com/adyen/checkout/card/CardView.kt

```diff
@@ -0,0 +1,318 @@
+                params.marginEnd = 0
+                binding.textInputLayoutExpiryDate.layoutParams = params
+            }
+        }
+        if (component.isStoredPaymentMethod() && component.requiresInput()) {
+            binding.textInputLayoutSecurityCode.editText?.requestFocus()
+        }
+    }
+
+    override fun observeComponentChanges(lifecycleOwner: LifecycleOwner) {
+        component.observeOutputData(lifecycleOwner, this)
+    }
+
+    override fun isConfirmationRequired(): Boolean {
+        return true
+    }
+
+    override fun highlightValidationErrors() {
+        component.outputData?.let {
+            var isErrorFocused = false
+            if (!it.cardNumberField.isValid) {
+                isErrorFocused = true
+                binding.editTextCardNumber.requestFocus()
+                setCardNumberError(R.string.checkout_card_number_not_valid)
+            }
+            if (!it.expiryDateField.isValid) {
+                if (!isErrorFocused) {
+                    isErrorFocused = true
+                    binding.textInputLayoutExpiryDate.requestFocus()
+                }
+                binding.textInputLayoutExpiryDate.error = mLocalizedContext.getString(R.string.checkout_expiry_date_not_valid)
+            }
+            if (!it.securityCodeField.isValid) {
+                if (!isErrorFocused) {
+                    isErrorFocused = true
+                    binding.textInputLayoutSecurityCode.requestFocus()
+                }
+                binding.textInputLayoutSecurityCode.error = mLocalizedContext.getString(R.string.checkout_security_code_not_valid)
+            }
+            if (binding.textInputLayoutCardHolder.visibility == VISIBLE && !it.holderNameField.isValid) {
+                if (!isErrorFocused) {
+                    binding.textInputLayoutCardHolder.requestFocus()
+                }
+                binding.textInputLayoutCardHolder.error = mLocalizedContext.getString(R.string.checkout_holder_name_not_valid)
+            }
+        }
+    }
+
+    private fun notifyInputDataChanged() {
+        component.inputDataChanged(mCardInputData)
+    }
+
+    private fun onCardNumberValidated(validatedNumber: ValidatedField<String>, detectedCardTypes: List<DetectedCardType>) {
+        if (validatedNumber.validation == ValidatedField.Validation.VALID) {
+            changeFocusOfInput(validatedNumber.value)
+        }
+        if (detectedCardTypes.isEmpty()) {
+            binding.cardBrandLogoImageView.setStrokeWidth(0f)
+            binding.cardBrandLogoImageView.setImageResource(R.drawable.ic_card)
+            binding.editTextCardNumber.setAmexCardFormat(false)
+        } else {
+            binding.cardBrandLogoImageView.setStrokeWidth(RoundCornerImageView.DEFAULT_STROKE_WIDTH)
+            mImageLoader?.load(detectedCardTypes[0].cardType.txVariant, binding.cardBrandLogoImageView)
+            // TODO: 29/01/2021 get this logic from OutputData
+            var isAmex = false
```

Comment:
You can use `val isAmex = detectedCardTypes.any { it.cardType == CardType.AMERICAN_EXPRESS }`